python/pyspark/sql/connect/streaming/query.py

```diff
@@ -66,11 +70,35 @@ def isActive(self) -> bool:
 
     isActive.__doc__ = PySparkStreamingQuery.isActive.__doc__
 
-    # TODO (SPARK-42960): Implement and uncomment the doc
+    def _execute_await_termination_cmd(self, timeout: int = 10) -> bool:
+        cmd = pb2.StreamingQueryCommand()
+        cmd.await_termination.timeout_ms = timeout
+        terminated = self._execute_streaming_query_cmd(cmd).await_termination.terminated
+        return terminated
+
+    def _await_termination(self, timeoutMs: Optional[int]) -> Optional[bool]:
```

Comment:
Something like this is probably better to do on the server side.
For this PR, I think we can skip doing this. We already have long running RPCs like 'processAllAvailable()'. We can handle them better together on the server side.